packages/mui-system/src/ThemeProvider/ThemeProvider.test.js

```diff
@@ -42,4 +42,182 @@ describe('ThemeProvider', () => {
     );
     expect(theme).to.include({ foo: 'foo' });
   });
+
+  it('merge theme by default', () => {
+    let theme;
+
+    function Test() {
+      theme = useTheme();
+
+      return null;
+    }
+    render(
+      <ThemeProvider theme={{ foo: 'foo' }}>
+        <ThemeProvider theme={{ bar: 'bar' }}>
+          <Test />
+        </ThemeProvider>
+      </ThemeProvider>,
+    );
+    expect(theme).to.deep.equal({ foo: 'foo', bar: 'bar', [Symbol.for('mui.nested')]: true });
```

Comment:
Does `[Symbol.for('mui.nested')]: true` mean that there's more than one `ThemeProvider`?

Reply:
Yes, nested providers.

```js
<ThemeProvider>
  <ThemeProvider>
  
  
// but not
<ThemeProvider>
<ThemeProvider>
```

Reply:
To be honest, I'm not sure what's the point of adding the symbol.

Reply:
Seems like it is for `@mui/styles` which is our legacy styled-engine.